### PR TITLE
chore(gpu): Add bitnot operation without using pbs

### PR DESCRIPTION
**Pull request by guillermo-oyarzun**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Review by agnesLeroy — COMMENTED** [review]
Hey! Thanks a lot for this PR! I see you have modified the content of `bitnot_assign_async`, but actually you should replace the content of `unchecked_bitnot_assign_async` with your changes. The unchecked version does not perform the initial carry propagation. :slightly_smiling_face:

**Comment by guillermo-oyarzun** [comment]
> Hey! Thanks a lot for this PR! I see you have modified the content of `bitnot_assign_async`, but actually you should replace the content of `unchecked_bitnot_assign_async` with your changes. The unchecked version does not perform the initial carry propagation. 🙂

ahh I followed the CPU implementation. On CPU there is no unchecked_bitnot, maybe we should not have too?

**Comment by agnesLeroy** [comment]
> > Hey! Thanks a lot for this PR! I see you have modified the content of `bitnot_assign_async`, but actually you should replace the content of `unchecked_bitnot_assign_async` with your changes. The unchecked version does not perform the initial carry propagation. 🙂
> 
> ahh I followed the CPU implementation. On CPU there is no unchecked_bitnot, maybe we should not have too?

@tmontaigu is there a reason there is no unchecked_bitnot on the CPU?

File: tfhe/src/integer/gpu/server_key/radix/bitwise_op.rs [hunk 1]
```diff
@@ -788,8 +792,27 @@ impl CudaServerKey {
...
+        // We do (-ciphertext) + (msg_mod -1) as it allows to avoid an allocation
+        cuda_lwe_ciphertext_negate_assign(&mut ct.as_mut().d_blocks, stream);
+
+        let ct_blocks = ct.as_ref().d_blocks.lwe_ciphertext_count().0;
+
+        let scalar = self.message_modulus.0 as u8 - 1;
+        let delta = (1_u64 << 63) / (self.message_modulus.0 * self.carry_modulus.0) as u64;
+        let shift_plaintext = u64::from(scalar) * delta;
+
+        let scalar_vector: Vec<u64> = std::iter::repeat(shift_plaintext).take(ct_blocks).collect();
```
Comment (tmontaigu) [inline tfhe/src/integer/gpu/server_key/radix/bitwise_op.rs:804]: `let scalar_vector = vec![shift_plaintext; count]`
should also work

**Comment by tmontaigu** [comment]
No I don't think there is a particular reason

**Comment by agnesLeroy** [comment]
Ok so @guillermo-oyarzun let's keep the unchecked entry point for the GPU and do the modifications there, I think it's better as it follows the same pattern as the other functions then.

**Review by agnesLeroy — APPROVED** [review]

**Pull request merged**